From: Holger Assmann <h.assmann@pengutronix.de>
Date: Thu, 17 Feb 2022 12:02:49 +0100
Subject: [PATCH] ARM: stm32mp: lxa-tac: introduce hardware watchdog

With iwdg2, the non-secure watchdog is used.
iwdg1 would be TrustZone-aware.

Signed-off-by: Holger Assmann <h.assmann@pengutronix.de>
---
 arch/arm/boot/dts/stm32mp157c-lxa-tac.dts | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/arch/arm/boot/dts/stm32mp157c-lxa-tac.dts b/arch/arm/boot/dts/stm32mp157c-lxa-tac.dts
index 4ffcd8283ec0..2e032af8c1ab 100644
--- a/arch/arm/boot/dts/stm32mp157c-lxa-tac.dts
+++ b/arch/arm/boot/dts/stm32mp157c-lxa-tac.dts
@@ -279,6 +279,11 @@ &rtc {
 	status = "okay";
 };
 
+&iwdg2 {
+	timeout-sec = <8>;
+	status = "okay";
+};
+
 &timers2 {
 	/* spare dmas for other usage */
 	/delete-property/dmas;
